[12.0][ADD] website_snippet_product_category (#360)
OCA/e-commerce: website_snippet_product_category/README.rst

    :target: https://odoo-community.org/page/development-status
    :alt: Beta
.. |badge2| image:: https://img.shields.io/badge/licence-LGPL--3-blue.png
    :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
    :alt: License: LGPL-3
.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fe--commerce-lightgray.png?logo=github
    :target: https://github.com/OCA/e-commerce/tree/12.0/website_snippet_product_category
    :alt: OCA/e-commerce
.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
    :target: https://translation.odoo-community.org/projects/e-commerce-12-0/e-commerce-12-0-website_snippet_product_category
    :alt: Translate me on Weblate
.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
    :target: https://runbot.odoo-community.org/runbot/113/12.0
    :alt: Try me on Runbot

|badge1| |badge2| |badge3| |badge4| |badge5| 

Adds a new snippet to show e-commerce categories

**Table of contents**

.. contents::
   :local:

Usage
=====

#. Go to website (frontend)
#. Press 'Edit' button
#. Drag the 'Product Category' snippet to the place of the website that you want
#. Save changes

> You can select what categories can be shown by the snippet and an image for the category:

Reviewer: This should go to `CONFIGURE.rst`.

#. Go to website (backend) > Configuration > Products > eCommerce Categories
#. Create or Edit one
#. You can see two new options "Website Published" and "Category Image"

Known issues / Roadmap
======================

> The snippet shows a maximum of 4 levels by design.

Reviewer: This is a point also

> #. Adds a option to configure the initial level

Reviewer: Don't use numbered list here. Use `*`

    :target: https://odoo-community.org/page/development-status
    :alt: Beta
.. |badge2| image:: https://img.shields.io/badge/licence-LGPL--3-blue.png
    :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
    :alt: License: LGPL-3
.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fe--commerce-lightgray.png?logo=github
    :target: https://github.com/OCA/e-commerce/tree/12.0/website_snippet_product_category
    :alt: OCA/e-commerce
.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
    :target: https://translation.odoo-community.org/projects/e-commerce-12-0/e-commerce-12-0-website_snippet_product_category
    :alt: Translate me on Weblate
.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
    :target: https://runbot.odoo-community.org/runbot/113/12.0
    :alt: Try me on Runbot

|badge1| |badge2| |badge3| |badge4| |badge5| 

Adds a new snippet to show e-commerce categories

**Table of contents**

.. contents::
   :local:

Configuration
=============

You can select what categories can be shown by the snippet and an image for the category:

> #. Go to website (backend) > Configuration > Products > eCommerce Categories

Reviewer: You have to put an empty line here for proper rendering

#. Create or Edit one
#. You can see two new options "Website Published" and "Category Image"

Usage
=====

#. Go to website (frontend)
#. Press 'Edit' button
#. Drag the 'Product Category' snippet to the place of the website that you want
#. Save changes

Theming
~~~~~~~

> * `.categ_container` > The container per master category

Reviewer: Say that these are `CSS classes`

* `.categ_tree_level` > The category row (image + text)
* `.main_tree_level` > The first category row level displayed
* `.categ_link` > The link of the category row
* `.categ_img` > The container of the icon of the category row

You can use 'data-tree-level' attribute to select the category row of a selected 
level (1-4 by design).

Also you can replace the entire template replacing the 'data-template' attribute value 
in the `s_product_category_options` view. You only need know that the name 
of the variable that have all the categories is called `object`.

Known issues / Roadmap
======================

* The snippet shows a maximum of 4 levels by design.

> * Adds a option to configure the initial level

Reviewer: ```suggestion
* Add an option to configure the initial level.
```

> * Adds a option to select how many levels can be represented

Reviewer: ```suggestion
* Add an option to select how many levels will be shown.
```
